app/controllers/jobs_controller.rb

```diff
@@ -21,24 +26,35 @@ def create
   end
 
   def show
+    @comment = Comment.new
     @job = Job.find(params[:id])
   end
 
   def edit
-    # implement on your own!
+    @categories = Category.all
+    @company = Company.find(params[:company_id])
+    @job = Job.find(params[:id])
   end
 
   def update
-    # implement on your own!
+    @company = Company.find(params[:company_id])
+    @job = Job.find(params[:id])
+    if Job.update(params[:id], job_params)
```

Comment:
we have the specific job object on the line above.. so we could just do `@job.update(job_params)`. this way is cleaner 👍